salt/utils/event.py

```diff
@@ -873,6 +882,36 @@ def __del__(self):
         except:  # pylint: disable=W0702
             pass
 
+    def _verify_event_schema(self, tag, event):
+
+        if not self.opts['validate_event_schema']:
+            return True
+
+        if not HAS_JSONSCHEMA:
+            raise salt.exceptions.SaltException("Jsonschema not found, schema validation should be turned off")
+
+        schema = self._find_schema_by_tag(tag)
+        try:
+            jsonschema.validate(event, schema.serialize())
+        except jsonschema.exceptions.ValidationError as exc:
+            if self.opts['raise_on_invalid_schema']:
+                raise salt.exceptions.InvalidSchema(exc)
+            else:
+                log.error("Unable to validate schema for tag %s" % tag)
+
+    def _find_schema_by_tag(self, tag):
+        all_schemas = []
+        for schema_mods in salt.loader.schemas(self.opts):
```

Comment:
We should not invoke the loader like this, this is causing the loader to re-scan with every event and will be a nightmare.
We should make an object level instance of the schema loader object and then use it.